### PR TITLE
error handling, solution, and testing for task 8 PATCH /api/articles/…

### PR DESCRIPTION
**Pull request by DeeSzubert**
error handling, solution, and testing for task 8 PATCH /api/articles/:article_id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -104,6 +104,32 @@ describe("/api/articles/:article_id", () => {
...
+  test("PATCH 200: Respond with updated article with votes more than 0", () => {
+    const newArticleVote = {
+      inc_votes: 20,
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(newArticleVote)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.votes).toBe(120);
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:116]: I think for this one, I'd like to see that you have actually returned everything! Usually, I would say to be concise, but this specifically says that it'll return the updated _article_, not just the votes :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -123,6 +149,35 @@ describe("/api/articles/:article_id", () => {
...
+    return request(app)
+      .patch("/api/articles/9999")
+      .send()
+      .expect(404)
+      .then(({ body }) => {
+        const { message } = body;
+        expect(message).toBe("id not found");
+      });
+  });
 });
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:181]: what if something is wrong with the object you are trying to send? I.e. inc_votes has a value of 'twenty'? There's also the thought that something could be added to that object - would that be considered an error?

File: controllers/articles-controllers.js [hunk 3]
```diff
@@ -18,4 +20,18 @@ function getArticles(request, response, next) {
...
+function getPatchedArticle(request, response, next) {
+  const { article_id } = request.params;
+  const { inc_votes } = request.body;
+
+  checkIfArticleExists(article_id)
+    .then((result) => {
+      return patchArticle(article_id, inc_votes);
+    })
+    .then((patchedArticle) => {
+      response.status(200).send(patchedArticle);
```
Comment (nc-niamh-smith) [inline controllers/articles-controllers.js:33]: I think this should be sent on an object :)

File: models/articles-models.js [hunk 4]
```diff
@@ -32,4 +32,20 @@ function checkIfArticleExists(article_id) {
...
+  return db
+    .query(
+      `UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *`,
+      [inc_votes, article_id]
+    )
+    .then(({ rows }) => {
+      return rows[0];
+    });
+}
+
```
Comment (nc-niamh-smith) [inline models/articles-models.js:45]: 👍

**Review by nc-niamh-smith — COMMENTED** [review]
Your logic is completely there! There are a couple of scenarios I'd like you to think about :)

**Comment by DeeSzubert** [comment]
Thank you Niamh

⁣Get BlueMail for Android ​

**Pull request merged**